### PR TITLE
AGENT-133: Move Ignition generation from fleeting code to asset

### PR DESCRIPTION
**Pull request by rwsu**
Instead of calling the fleeting code to generate the Ignition config, the Ignition asset now builds it directly.

To make it easier for core reviewers, the first commit touches core installer code and has been submitted as its own PR #5944.

~~Depends on: #5943~~
~~Depends on: #5944~~

### CODE REVIEW

File: pkg/asset/agent/image/ignition.go [hunk 1]
```diff
@@ -31,16 +74,107 @@ func (a *Ignition) Generate(dependencies asset.Parents) error {
...
+		return fmt.Errorf("pull secret is missing")
+	}
+
+	nodeZeroIP, err := manifests.GetNodeZeroIP(agentManifests.NMStateConfigs)
+	if err != nil {
+		return err
+	}
+
+	agentTemplateData := getTemplateData(
+		pullSecret.StringData[".dockerconfigjson"],
```
Comment (andfasano) [inline pkg/asset/agent/image/ignition.go:106]: Could be retrieved from the method added in [PR 5943](https://github.com/openshift/installer/pull/5943/files#diff-ed3ddff72757f2b9f8078d820e70ac801ad04c31885e033d6722ac6ae865f10eR92)

**Comment by andfasano** [comment]
In general the approach sounds good to me, I think anyhow this PR should follow https://github.com/openshift/installer/pull/5943, also to reuse some of the validations/tests introduced (not yet present here, but I guess since it's still a WIP)

**Comment by bfournie** [comment]
This shouldn't hold this up but it looks like the unit test for GetNodeZeroIP() from here https://github.com/openshift/installer/blob/agent-installer/pkg/agent/manifests/nmstateconfig_test.go#L10-L89 got dropped. 
This can be added back in a follow-on.

/lgtm

**Comment by bfournie** [comment]
I verified this with a successful install in dev-scripts using:
`export INSTALLER_REPO_PR=5945`

/approve

**Pull request merged**